service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/PaymentServiceTest.java

```diff
@@ -93,18 +96,21 @@ void setup() {
             when(feeService.getFeesDataForC2(WITHOUT_NOTICE)).thenReturn(buildFeesData(feeForC2WithoutNotice));
         }
 
-        @Test
-        void shouldMakeCorrectPaymentForC2WithNotice() {
+        @ParameterizedTest
+        @NullAndEmptySource
+        @ValueSource(strings = {"customerReference"})
+        void shouldMakeCorrectPaymentForC2WithNotice(final String customerReference) {
             CaseData caseData = CaseData.builder()
                 .caseLocalAuthority("LA")
                 .c2DocumentBundle(List.of(element(C2DocumentBundle.builder()
                     .type(WITH_NOTICE)
                     .pbaNumber("PBA123")
                     .clientCode("clientCode")
-                    .fileReference("fileReference")
+                    .customerReference(customerReference)
                     .build())))
                 .build();
-            CreditAccountPaymentRequest expectedPaymentRequest = testCreditAccountPaymentRequestBuilder()
+            CreditAccountPaymentRequest expectedPaymentRequest = testCreditAccountPaymentRequestBuilder(
+                customerReference)
```

Comment:
I don't see how this test works, if we pass nullandemptySource why we are not explicitly checking against "Not provided" is different label.